ui/src/app/pages/analysis/notebook-redirect.tsx

```diff
@@ -0,0 +1,453 @@
+]);
+
+const ProgressCard: React.FunctionComponent<{currentState: Progress, index: ProgressCardState,
+  progressComplete: Map<Progress, boolean>, creatingNewNotebook: boolean}> =
+  ({currentState, index, progressComplete, creatingNewNotebook}) => {
+    const includesStates = progressCardStates.get(index);
+    const isCurrent = includesStates.includes(currentState);
+    const isComplete = includesStates.every(s => s.valueOf() < currentState.valueOf());
+
+    // Conditionally render card text
+    const renderText = () => {
+      switch (index) {
+        case ProgressCardState.UnknownInitializingResuming:
+          if (currentState === Progress.Unknown || progressComplete[Progress.Unknown]) {
+            return 'Connecting to the notebook server';
+          } else if (currentState === Progress.Initializing ||
+            progressComplete[Progress.Initializing]) {
+            return 'Initializing notebook server, may take up to 10 minutes';
+          } else {
+            return 'Resuming notebook server, may take up to 1 minute';
+          }
+        case ProgressCardState.Authenticating:
+          return 'Authenticating with the notebook server';
+        case ProgressCardState.CopyingCreating:
+          if (creatingNewNotebook) {
+            return 'Creating the new notebook';
+          } else {
+            return 'Copying the notebook onto the server';
+          }
+        case ProgressCardState.Redirecting:
+          return 'Redirecting to the notebook server';
+      }
+    };
+
+    const icon = progressCardIcons.get(index);
+    return <div style={isCurrent ? {...styles.progressCard, backgroundColor: '#F2FBE9'} :
+      styles.progressCard}>
+      {isCurrent ? <Spinner style={{width: '46px', height: '46px'}}
+                            data-test-id={'progress-card-spinner-' + index.valueOf()}/> :
+        <React.Fragment>
+          {icon.shape === 'notebook' ? <NotebookIcon style={styles.progressIcon}/> :
+          <ClrIcon shape={icon.shape} style={isComplete ?
+          {...styles.progressIcon, ...styles.progressIconDone,
+            transform: icon.rotation} :
+            {...styles.progressIcon, transform: icon.rotation}}/>}
+        </React.Fragment>}
+        <div style={styles.progressText}>
+          {renderText()}
+        </div>
+    </div>;
+  };
+
+interface State {
+  initialized: boolean;
+  leoUrl: string;
+  localizationError: boolean;
+  progress: Progress;
+  progressComplete: Map<Progress, boolean>;
+}
+
+interface Props {
+  workspace: WorkspaceData;
+  queryParams: any;
+  profileState: {profile: Profile, reload: Function, updateCache: Function};
+}
+
+export const NotebookRedirect = fp.flow(withUserProfile(), withCurrentWorkspace(),
+  withQueryParams())(class extends React.Component<Props, State> {
+
+    private timeoutReference: NodeJS.Timer;
+    private aborter = new AbortController();
+
+    constructor(props) {
+      super(props);
+      this.state = {
+        initialized: false,
```

Comment:
nit: Existing issue that I probably personally introduced originally, but rereading this I would probably make this slightly more specific: `clusterInitialized`. `initialized` at first glance could be misinterpreted as some kind of overall component initialization state.

Reply:
removed "initialized" from state - it now initializes in `initializeClusterStatusChecking()` which is called before this